nilearn/version.py

```diff
@@ -102,6 +100,42 @@ def _import_module_with_version_check(
     return module
 
 
+def _compare_version(version_a, operator, version_b):
+    """Compare two version strings via a user-specified operator.
+
+    ``distutils`` has been deprecated since Python 3.10 and is scheduled
+    for removal from the standard library with the release of Python 3.12.
+    For version comparisons, we use setuptools's `parse_version` if available.
+
+    Note : function borrowed from the MNE team.
+
+    Parameters
+    ----------
+    version_a : :obj:`str`
+        First version string.
+
+    operator : {'==', '>', '<', '>=', '<='}
+        Operator to compare ``version_a`` and ``version_b`` in the form of
+        ``version_a operator version_b``.
+
+    version_b : :obj:`str`
+        Second version string.
+
+    Returns
+    -------
+    result : :obj:`bool`
+        The result of the version comparison.
+
+    """
+    try:
+        from pkg_resources import parse_version as parse  # noqa:F401
+    except ImportError:
+        from distutils.version import LooseVersion as parse  # noqa:F401
+    return eval(
+        'parse("{0}") {1} parse("{2}")'.format(version_a, operator, version_b)
```

Comment:
In general using `eval` (esp. on unvalidated input) is not a very good habit, could we use a mapping from strings to functions in the standard library module [`operator`](https://docs.python.org/3/library/operator.html) instead?

Reply:
Good point indeed, thanks @jeromedockes !
I added a mapping in 3c6e888